libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -116,6 +116,20 @@ SourceLocation ASTJsonImporter::createNameSourceLocation(Json::Value const& _nod
 	return solidity::langutil::parseSourceLocation(_node["nameLocation"].asString(), m_sourceNames);
 }
 
+SourceLocation ASTJsonImporter::createKeyNameSourceLocation(Json::Value const& _node)
+{
+	astAssert(member(_node, "keyNameLocation").isString(), "'keyNameLocation' must be a string");
+
+	return solidity::langutil::parseSourceLocation(_node["keyNameLocation"].asString(), m_sourceNames);
+}
+
+SourceLocation ASTJsonImporter::createValueNameSourceLocation(Json::Value const& _node)
+{
+	astAssert(member(_node, "valueNameLocation").isString(), "'valueNameLocation' must be a string");
+
+	return solidity::langutil::parseSourceLocation(_node["valueNameLocation"].asString(), m_sourceNames);
+}
+
```

Comment:
A bit of not-exactly-necessary duplication going on here, but I'll let it slide...

Reply:
Ah yes, another thing I mentally noted down to ask about, and then of course, forgot immediately. This should be fairly easy to simplify into one helper function, but as we're gonna be moving this to `VariableDeclaration` at a certain point (once we accumulate enough changes), I'm wondering if it's worthwhile to do now?

Reply:
Yeah, it's fine for now - but yeah, we'll need to see and coordinate with tooling how and when exactly we'd make such a transition - ideally we'd have some external AST translation tool collectively maintained by all tooling with our help, to avoid having duplicated version-dependent workarounds everywhere, before we start changing the AST too aggressively. In principle, it should be easy to forward-compatibly translate from the current format to a variable declaration. But yeah, let's see. Anyways, as I said: fine for now.